### ISSUE TITLE
Support HTTP authentication for git dependencies

### ISSUE THREAD

**Issue by evgenii-petrov-arrival** [python-poetry/poetry#2050]
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] I have searched the [documentation](https://python-poetry.org/docs/) and believe that my question is not covered.

## Feature Request
I want to pull a dependency from a private git repo, but the repo requires HTTP authentication.

As far as I can tell, https://python-poetry.org/docs/repositories/ only supports PyPI repositories, so git repositories are not supported.

I want to be able to do something similar to:

```
$ poetry config repositories.git-domain-repo https://git.domain/repo/
$ poetry config http-basic.git-domain-repo username password
```

Or, even better, to configure auth for the whole `git.domain`.

Please add a way to specify a username and a password to be used when pulling a git dependency, such that the username and password are specified outside of pyproject.toml .

**Comment by ephes** [comment]
Poetry is a really cool project. This is my first issue using it - so all thumbs up :).

I'm not sure this should be a feature request. Stumbled across this trying to migrate a project from requirements.txt to poetry. Pip installed the git url with http basic authentication, but poetry did not fail showing an error message, it just ran into an endless dependency resolving loop:
```
Updating dependencies
Resolving dependencies... (276.9s)^C
```
I think this should be a bug report.

**Comment by finswimmer** [comment]
@ephes: The problem in your example is the `+` in the username. This will be fixed with PR #2169.

fin swimmer

**Comment by ephes** [comment]
Thanks a lot, I suspected something like that and therefore preserved the „+“ :).

**Comment by lpilz** [comment]
@evgenii-petrov-arrival I second this feature request, however this might be solved with #5581. Not 100% sure though

**Comment by abn** [comment]
HTTP Authentication is already supported in Poetry. See #5567 #5581.

Doc: https://python-poetry.org/docs/dependency-specification/#git-dependencies
